azext_edge/edge/providers/check/base/node.py

```diff
@@ -0,0 +1,162 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from kubernetes.client.exceptions import ApiException
+from kubernetes.client.models import V1Node, V1NodeList
+from rich.padding import Padding
+from rich.table import Table
+from typing import Any, Dict
+
+from .check_manager import CheckManager
+from .user_strings import MULTINODE_CLUSTER_MSG, NO_NODES_MSG, UNABLE_TO_FETCH_NODES_MSG
+from ..common import (
+    AIO_SUPPORTED_ARCHITECTURES,
+    COLOR_STR_FORMAT,
+    DISPLAY_BYTES_PER_GIGABYTE,
+    MIN_NODE_MEMORY,
+    MIN_NODE_STORAGE,
+    MIN_NODE_VCPU
+)
+from ....common import CheckTaskStatus
+
+
+logger = get_logger(__name__)
+
+
+def check_nodes(as_list: bool = False) -> Dict[str, Any]:
+    from ...base import client
+    check_manager = CheckManager(check_name="evalClusterNodes", check_desc="Evaluate cluster nodes")
+    padding = (0, 0, 0, 8)
+    target = "cluster/nodes"
+    check_manager.add_target(target_name=target, conditions=["len(cluster/nodes)>=1"])
```

Comment:
This might just be a semantic nit but is our target `>= 1`? If we warn for having more than one node, maybe our target should be `==1`

Reply:
I thought we should still continue with checks if there is more than one node... plus a warning isn't a "fail" so the condition is still successful? maybe?